lib/copy/copy.js

```diff
@@ -2,51 +2,213 @@
 
 const fs = require('graceful-fs')
 const path = require('path')
-const ncp = require('./ncp')
-const mkdir = require('../mkdirs')
+const mkdirs = require('../mkdirs').mkdirs
+const utimes = require('../util/utimes').utimesMillis
 
-function copy (src, dest, options, callback) {
-  if (typeof options === 'function' && !callback) {
-    callback = options
+const DEST_NOENT = -1
+const DEST_EXISTS = 1
```

Comment:
I like the idea of using `Symbol`s here as one of their use cases is for unique constants.